anypay/anypay
# Measure http request response times with @promster/hapi

## Pull Request: Measure HTTP Request Response Times with @promster/hapi

### Overview

This pull request introduces the integration of `@promster/hapi` to measure and expose HTTP request response times within our Hapi.js application. Monitoring response times is crucial for understanding the performance of our server and identifying potential bottlenecks.

### Changes Made

- **Integration of @promster/hapi**: Added the necessary configurations and dependencies to seamlessly integrate `@promster/hapi` into our Hapi.js project.

- **Instrumentation for Response Times**: Modified the relevant parts of our codebase to capture and record HTTP request response times using the capabilities provided by `@promster/hapi`.

### Why This Matters

- **Performance Insights**: With this integration, we gain valuable insights into the performance of our HTTP routes, allowing us to identify and address potential performance issues.

- **Real-time Monitoring**: The exposed metrics can be monitored in real-time, enabling us to detect anomalies and proactively optimize our application.

### How to Test

To validate the changes, follow these steps:

1. **Clone the Repository**:

    ```bash
    git clone <repository-url>
    cd <repository-directory>
    ```

2. **Install Dependencies**:

    ```bash
    npm install
    ```

3. **Run the Application**:

    ```bash
    npm start
    ```

4. **Send HTTP Requests**:

    Use tools like `curl` or your preferred API testing tool to send HTTP requests to the server.

5. **Check Metrics Endpoint**:

    Visit the metrics endpoint (e.g., `/metrics`) to view the recorded response times.

### Additional Notes

- Ensure that the appropriate configuration options for `@promster/hapi` are set up in our environment variables or configuration files.

- This integration aligns with our goal of maintaining a high-performance and reliable application.

### Related Issues

Closes: #<issue-number>

### Checklist

- [x] Code follows our coding standards
- [x] Documentation reflects the changes made
